api/src/main/scala/za/co/absa/loginsvc/rest/SecurityConfig.scala

```diff
@@ -57,29 +63,34 @@ class SecurityConfig @Autowired()(authConfigsProvider: AuthConfigProvider) {
       .sessionManagement()
         .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
         .and()
-      .httpBasic()
-
-    if(ldapConfig != null)
-      {
-        if(ldapConfig.enableKerberos.isDefined)
-        {
-          val kerberos = new KerberosSPNEGOAuthenticationProvider(ldapConfig)
+      // like "httpBasic", but with special handling fo custom exceptions:
+      .addFilterAt(new BasicAuthenticationFilter(authManager, customAuthenticationEntryPoint), classOf[BasicAuthenticationFilter])
```

Comment:
Without this filter, the entrypoint authentication was was only able to see `InsufficientAuthenticationException`, never any of the custom ones